### PR TITLE
Upgrade Consent Management Platform to version 9.0.1

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
Upgrades consent management platform to version unified Sourcepoint CDK - changes described in https://github.com/guardian/consent-management-platform/pull/485

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes https://github.com/guardian/dotcom-rendering/pull/3561

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Tested

- [X] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by mxdvl — COMMENTED** [review]
🚢

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "@guardian/automat-modules": "^0.3.8",
     "@guardian/braze-components": "^4.1.0",
     "@guardian/commercial-core": "0.24.x",
-    "@guardian/consent-management-platform": "6.11.5",
+    "@guardian/consent-management-platform": "9.x.x",
```
Comment (mxdvl) [inline package.json:23]: Maybe we want to ensure that we require at least 9.0.1, as 9.0.0 is broken!

```suggestion
    "@guardian/consent-management-platform": "^9.0.1",
```

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts [hunk 2]
```diff
@@ -165,10 +165,16 @@ export const init = (): Promise<void> => {
 		// A promise error here, from a failed module load,
 		// could be a network problem or an intercepted request.
 		// Abandon the init sequence.
-		setupAdvertising().then(adFreeSlotRemove).catch(removeSlots);
+		setupAdvertising()
+			.then(adFreeSlotRemove)
+			.catch((e) => {
+				console.log('will remove slots', e);
+				void removeSlots();
+			});
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts:173]: Not sure we need all these logs.

File: yarn.lock [hunk 3]
```diff
@@ -1908,12 +1908,12 @@
...
-  version "6.11.5"
-  resolved "https://registry.yarnpkg.com/@guardian/consent-management-platform/-/consent-management-platform-6.11.5.tgz#93681cbf61de0fadb4b65d835d7fc445cb847b31"
-  integrity sha512-ap7b09qk8I/EfoWB3NR8kT63hBpm4ATFGpRicCQ61DLMw2PgCWunMvizdbSmo5wNCQKPFgmX9cfRT5IGfPneKw==
+"@guardian/consent-management-platform@9.x.x":
+  version "9.0.1"
+  resolved "https://registry.yarnpkg.com/@guardian/consent-management-platform/-/consent-management-platform-9.0.1.tgz#bfcfca3eb32a65faa0f60ff723a294ba840d9953"
+  integrity sha512-ZN4tf7VpGxQu1CcqRsHNzWK61Q3KlaLGf8HKEX2oGotKzWZdoCx22tEA/v5Stdlv1ZkcMSqnfmtF1fJ5STgWGg==
   dependencies:
-    "@guardian/libs" "^1"
+    "@guardian/libs" "1.6.1 - 3.3.0"
```
Comment (mxdvl) [inline yarn.lock:1916]: Maybe we want to allow all versions of @guardian/libs v3. For the other repo.

**Review by mxdvl — COMMENTED** [review]
Some general comments about more precise NPM package versioning.

File: yarn.lock (hunk 3 above)
Comment (zekehuntergreen) [reply yarn.lock:1916]: Can't we let yarn work out what version it needs?

**Review by kenoir — APPROVED** [review]

**Pull request merged**